sql/core/src/main/scala/org/apache/spark/sql/jdbc/MySQLDialect.scala

```diff
@@ -259,6 +311,8 @@ private case class MySQLDialect() extends JdbcDialect with SQLConfHelper with No
     case StringType => Option(JdbcType("LONGTEXT", java.sql.Types.LONGVARCHAR))
     case ByteType => Option(JdbcType("TINYINT", java.sql.Types.TINYINT))
     case ShortType => Option(JdbcType("SMALLINT", java.sql.Types.SMALLINT))
+    // We override getJDBCType so that DoubleType is mapped to DOUBLE instead.
+    case DoubleType => Option(JdbcType("DOUBLE", java.sql.Types.DOUBLE))
```

Comment:
This change is to make the use of `DOUBLE` which is a synonym of `DOUBLE PRECISION`. But it is compatible in cast function from MySQL 5.7, which is still widely used, all the way up to the latest 9.1.